api/v1alpha7/openstackcluster_types.go

```diff
@@ -162,6 +162,12 @@ type OpenStackClusterStatus struct {
 	// externalNetwork contains information about the external network used for default ingress and egress traffic.
 	ExternalNetwork *NetworkStatus `json:"externalNetwork,omitempty"`
 
+	// router describes the default cluster router
+	Router *Router `json:"router,omitempty"`
+
+	// apiServerLoadBalancer describes the api server load balancer if one exists
```

Comment:
Comment case should match variable case `Router`, `ExternalNetwork`, `APIServerLoadBalancer` and end with a dot.

The dot linter would be great but is unfortunately not reliable enough.

Reply:
Interesting. I have no strong personal preference, but as our API astronauts in OpenShift always insist on using the json tag in API godocs I was assuming it was a documented convention. And it is, but the only reference I can find is for OpenShift specifically: https://github.com/openshift/enhancements/blob/master/dev-guide/api-conventions.md#use-json-field-names-in-godoc

The reasoning is that users see this output in `kubectl explain`, and only the json tags make sense in that context.

But... this isn't documented in kubernetes 🤔 The [API Conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md) don't have anything to say on the matter, and the [coding conventions](https://www.kubernetes.dev/docs/guide/coding-convention/) just point to the Go conventions, which obviously say to use the Go field. I'm genuinely surprised by that. I was kinda hoping to do a big API cleanup and change it everywhere so it's consistent, but now I'm wondering.

Thoughts?

Reply:
I think we don't even have to start with the object name itself. If we take a look at `kubectl explain` we can see that the name will be printed anyway.

Example

```
❯ k explain osc.spec | grep -A1 "^  network"
  network	<Object>
    If NodeCIDR cannot be set this can be used to detect an existing network.
```

If we want to keep the name in the comment I'd suggest to use the go notation and not the json tag.
But tbh, I don't really have a hard opinion on this.

Reply:
On that there is a convention: https://www.kubernetes.dev/docs/guide/coding-convention/ points to https://go.dev/blog/godoc, which says:

```
Notice this comment is a complete sentence that begins with the name of the element it describes. This important convention allows us to generate documentation in a variety of formats, from plain text to HTML to UNIX man pages, and makes it read better when tools truncate it for brevity, such as when they extract the first line or sentence.
```